source/style-guide/style/titles-and-headings.txt

```diff
@@ -276,9 +279,11 @@ contents (TOC) for a collection of content:
 -  Entries in the TOC should link only to sections in the content.
    Don't include a link to an outside resource in the TOC.
 
--  The text of a TOC entry must match the text of the title or heading
-   to which it links. If the link needs to be shorter, revise the
-   title or heading to be shorter.
+-  The text of a TOC entry should match the text of the title or
+   heading to which it links. If the link needs to be shorter, consider
+   shortening the title or heading or providing a more concise title or
+   heading for the TOC. An alternate TOC title or heading should convey
+   the same intent of the full title or heading.
```

Comment:
Might this be a good place to bring up renaming anchor links i.e. `:ref:` links where I change the name of the link away from its default header text? Example:

This FAQ item is very helpful:

https://docs.mongodb.com/master/faq/concurrency/#what-locks-are-taken-by-some-common-client-operations

But the question format isn't really appropriate for use in reference material. So I would probably do something like this:

> See the :ref:`list of common client operations requiring locks`<that_ref> for more information.

Reply:
(this is for refs, not the TOC, so different application, but similar approach.

Reply:
Added an example.